### PR TITLE
feat(profile): add flat update profile

### PR DESCRIPTION
**Pull request by stpnvntn**
**🟢 Add `deploy` label if you want to deploy this Pull Request to staging environment**

#### 🧑‍⚖️ Pull Request Naming Convention

- Title should follow [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary)
- Do not put issue id in title
- Do not put WIP in title. Use [Draft PR](https://github.blog/2019-02-14-introducing-draft-pull-requests/) functionality
- Consider to add `area:*` label(s)

* [x] I followed naming convention rules

---

#### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Performance optimization
- [ ] Refactoring
- [ ] Test Case
- [ ] Documentation update
- [ ] Other

#### 🔗 Related issue link

Fixes https://github.com/rolling-scopes/rsschool-app/issues/1637

#### 💡 Background and solution

_Describe the big picture of your changes here_

#### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Database migration is added or not needed
- [x] Documentation is updated/provided or not needed
- [x] Changes are tested locally

### CODE REVIEW

File: client/src/api/api.ts [hunk 1]
```diff
@@ -2307,6 +2307,109 @@ export interface ProfileInfoDto {
...
+ * @export
+ * @interface ProfileInfoFlatDto
+ */
+export interface ProfileInfoFlatDto {
+    /**
+     * 
+     * @type {GeneralInfo}
+     * @memberof ProfileInfoFlatDto
+     */
+    'generalInfo': GeneralInfo;
```
Comment (ShekoMaxim) [inline client/src/api/api.ts:2321]: You duplicate fields from `GeneralInfo` interface, make sense to do this?

Comment (stpnvntn) [reply client/src/api/api.ts]: Definitely not, thanks!

File: nestjs/src/profile/profile.service.ts [hunk 2]
```diff
@@ -135,6 +136,55 @@ export class ProfileService {
...
+      contactsEpamEmail,
+    } = profileInfo;
+
+    const [firstName, lastName] = name?.split(' ') ?? [];
+
+    const user = await this.userRepository
+      .createQueryBuilder()
+      .update(User)
+      .set(
+        getValuesToUpdate<User>({
```
Comment (apalchys) [inline nestjs/src/profile/profile.service.ts:163]: maybe we can just do `_.omitBy(object, _.isUndefined);` ?

File: nestjs/src/profile/profile.service.ts [hunk 3]
```diff
@@ -135,6 +136,55 @@ export class ProfileService {
...
+      contactsNotes,
+      contactsSkype,
+      contactsLinkedIn,
+      contactsEpamEmail,
+    } = profileInfo;
+
+    const [firstName, lastName] = name?.split(' ') ?? [];
+
+    const user = await this.userRepository
+      .createQueryBuilder()
```
Comment (apalchys) [inline nestjs/src/profile/profile.service.ts:160]: i think you can just use `this.userRepository.update(id, {...});` ?

Comment (stpnvntn) [reply nestjs/src/profile/profile.service.ts]: Sure! Thanks

**Review by apalchys — APPROVED** [review]

**Comment by apalchys** [comment]
@I-vasilich-I could you please check if it's what you need?

File: nestjs/src/profile/dto/update-profile.dto.ts [hunk 4]
```diff
@@ -234,3 +234,80 @@ export class ProfileInfoDto {
...
+
+  @ApiProperty({ required: false, nullable: true, type: String })
+  @IsOptional()
+  @IsString()
+  contactsPhone?: string | null;
+
+  @ApiProperty({ required: false, nullable: true, type: String })
+  @IsOptional()
+  @IsString()
+  contactsEmail: string | null;
```
Comment (I-vasilich-I) [inline nestjs/src/profile/dto/update-profile.dto.ts:281]: should it be `contactsEmail?: string | null;` like everything else?

**Comment by I-vasilich-I** [comment]
> @I-vasilich-I could you please check if it's what you need?

@apalchys Looks like it is

**Comment by apalchys** [comment]
@stpnvntn  please rebase and feel free to merge

File: nestjs/src/profile/dto/update-profile.dto.ts (hunk 4 above)
Comment (stpnvntn) [reply nestjs/src/profile/dto/update-profile.dto.ts]: thanks, fixed

**Pull request merged**